### PR TITLE
Make ServiceClient Message not disposable

### PR DESCRIPTION
**Pull request by tmahmood-microsoft**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `main` branch.

### CODE REVIEW

File: iothub/service/src/Messaging/Models/Message.cs [hunk 1]
```diff
@@ -259,217 +161,81 @@ public string ContentEncoding
...
-
-        internal bool HasBodyStream()
-        {
-            return _bodyStream != null;
-        }
-
-        /// <summary>
-        /// Gets or sets the deliveryTag which is used for server side checkpointing.
-        /// </summary>
-        internal ArraySegment<byte> DeliveryTag { get; set; }
```
Comment (drwill-ms) [inline iothub/service/src/Messaging/Models/Message.cs:281]: Did we remove this because it was unused?

File: iothub/service/tests/MessageTests.cs [hunk 2]
```diff
@@ -19,126 +20,20 @@ public class MessageTests
...
-        [TestMethod]
-        public void HttpExceptionMappingTest_BulkRegistryOperationFailure()
-        {
-            var message = new HttpResponseMessage(HttpStatusCode.BadRequest);
-            message.Headers.Add(CommonConstants.IotHubErrorCode, ErrorCode.BulkRegistryOperationFailure.ToString());
-            bool isMappedToException = HttpClientHelper.IsMappedToException(message);
-            Assert.IsFalse(isMappedToException, "BulkRegistryOperationFailures should not be mapped to exceptions");
+            var msg = new Message(new byte[0]);
+            Assert.IsNotNull(msg.Payload);
+            Assert.AreEqual(0, msg.Payload.Length);
```
Comment (drwill-ms) [inline iothub/service/tests/MessageTests.cs:36]:
```suggestion
            msg.Payload.Should().NotBeNull();
            msg.Payload.Length.Should().Be(0);
```

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/Messaging/Models/Message.cs (hunk 1 above)
Comment (tmahmood-microsoft) [reply iothub/service/src/Messaging/Models/Message.cs:281]: Did not remove it, just moved it to the bottom of the class.

Comment (drwill-ms) [reply iothub/service/src/Messaging/Models/Message.cs:281]: Because internal? Makes sense.

I've tried looking for it, but I can't find it. Line?

Comment (drwill-ms) [reply iothub/service/src/Messaging/Models/Message.cs:281]: Never mind. I see it.

**Pull request merged**